tests/forms/test_dmfields.py

```diff
@@ -73,13 +73,37 @@ class TestForm(wtforms.Form):
 
     form = TestForm()
 
-    assert form.field.id == "input-field"
+    assert form.field.id.startswith("input-")
 
 
 def test_field_id_is_not_prefixed_with_input_prefix_if_specified(field_class):
     class TestForm(wtforms.Form):
-        field = field_class(id="input-field")
+        field = field_class(id="field")
 
     form = TestForm()
 
-    assert form.field.id == "input-field"
+    assert form.field.id == "field"
+
+
+@pytest.mark.parametrize("field_class", (
+    dmutils.forms.fields.DMRadioField, dmutils.forms.fields.DMBooleanField
+))
+def test_selection_field_id_is_suffixed_with_number(field_class):
+    class TestForm(wtforms.Form):
+        field = field_class()
+
+    form = TestForm()
+
+    assert form.field.id == "input-field-1"
```

Comment:
Can we test that DMRadioFields with more than one option don't all have suffix `-1`?

Reply:
They won't have the suffix `-1`... However because WTForms already suffixes a number each option will end up having the an id `input-field-1-{i}` 😅 

However, in our codebase we don't use the WTForms options for generating HTML, so I don't think those `id`s get used anywhere.

So this implementation is a hack, because we're relying on that assumption.

However, I can't think of a way to implement the correct behaviour of returning the id of the first option that isn't more broken. Any ideas?

Reply:
I've tried to circumvent the issue by adding a new property to the field objects, `href`. I've kept it as a separate commit for now but it can get folded in if you think it's better.